packages/core/directives/control.ts

```diff
@@ -0,0 +1,44 @@
+import {Component, ElementRef, Input, OnChanges, SimpleChanges, ViewEncapsulation} from '@angular/core';
+import {GoogleMapsAPIWrapper} from '../services/google-maps-api-wrapper';
+import {ControlPosition} from '../services/google-maps-types';
+
+/**
+ * AgmMapControl allows to add a custom control to the map
+ *
+ * ### Example
+ *
+ * ```
+ * <agm-control [position]="position" #my-id>
+ *   <div content>
+ *       <!-- my markup -->
+ *   </div>
+ * </agm-control>
+ * ```
+ *
+ */
+@Component({
+  selector: 'agm-control',
+  template: '<ng-content select="[content]"></ng-content>',
+  styles:
+      ['.agm-control {' +
+       '  background: white;' +
+       '  padding: 10px;' +
+       '}'],
+  encapsulation: ViewEncapsulation.None,
+  host: {class: 'agm-control'}
+})
+export class AgmControl implements OnChanges {
+  /* Position of the control */
+  @Input() position: ControlPosition;
+
+  constructor(private elm: ElementRef, private _mapsWrapper: GoogleMapsAPIWrapper) {}
+
+  /* @internal */
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['position'] && this.position) {
```

Comment:
when the position changes from filled to empty, we should remove the control here also

Reply:
Done at: https://github.com/SebastianM/angular-google-maps/pull/1398/files#diff-6196ef893a3ec1882bf45cc192360ea5R39